web/themes/custom/jcc_components/includes/views.inc

```diff
@@ -393,3 +393,140 @@ function jcc_components_preprocess_views_view__documents__default(&$variables) {
+
+      default:
+        break;
+    }
+    if (isset($extend_rss)) {
+      // Content type.
+      $variables['content_type'] = $content_type;
+      // "Sub-type" tag.
+      if (isset($tid)) {
+        $term = Term::load($tid);
+        if (isset($term)) {
+          $variables['type'] = $term->getName();
+        }
+      }
+      // Author, not page creator, when applicable such as news contact.
+      if (isset($node->field_contact) && !empty($node->get('field_contact')->target_id)) {
+        $author_id = $node->get('field_contact')->target_id;
+        $variables['author'] = Node::load($author_id);
+      }
+      // Tagline.
+      if ($node->hasField('field_tagline')) {
+        $variables['tagline'] = $node->get('field_tagline')->value;
+      }
+      // Publishing date for news article.
+      $variables['pubDate'] = date('F j, Y', $node->created->value);
+      // Date range when applicable such an event.
+      if (isset($node->field_date_range) && !empty($node->get('field_date_range'))) {
+        $variables['startDate'] = $node->get('field_date_range')->value;
+        $variables['endDate'] = $node->get('field_date_range')->end_value;
+      }
+      // Body and teaser.
+      $variables['body'] = $node->body->view('full');
+      $variables['teaser'] = $node->body->view('teaser');
+      // 1.x Extended event content to concat with main body.
+      if ($content_type == 'Event' && !empty($node->get('field_event_moreinfo'))) {
+        $variables['extended_details'] = $node->field_event_moreinfo->view('full');
+      }
+      // Attach images not inline with body.
+      if (isset($node->field_images) && count($node->field_images) != 0) {
+        $first_image = _jcc_get_first_image($node);
+        $variables['image'] = $first_image['url'];
+      }
+      // Publish status.
+      $variables['pubStatus'] = $node->get('status')->getString();
+    }
+  }
+
+}
+
+/**
+ * @file
+ * Preprocessors for views.
+ */
+
+/**
+ * Implements hook_preprocess_HOOK().
+ */
+function jcc_components_preprocess_views_view__feeds_directory__page_1(&$variables) {
+  if (empty($variables['rows'])) {
+    return;
+  }
+  $feeds_types = [];
+  $feeds_topics = [];
+  foreach ($variables['rows'][0]['#rows'] as $row) {
+    $term = $row['#row']->_entity;
+    $term_name = $term->name->value;
+    $tid = $term->tid->value;
+    $vid = Term::load($tid)->bundle();
+    $feed_details = [
+      'icon' => [
+        'name' => 'rss2',
+        'title' => 'rss2',
+      ],
+      'name' => $term_name,
+      'url' => '/cats/' . $tid,
+    ];
+
+    switch ($vid) {
+      case 'news_type':
+        array_push($feeds_topics, $feed_details);
+        break;
+
+      default:
+        array_push($feeds_types, $feed_details);
+    }
+  }
+
+  $feeds_types = $feeds_topics;
+  foreach ($feeds_types as $key => $type) {
+    if ($type['name'] != 'News Release') {
+      unset($feeds_types[$key]);
+    }
+  }
+  $feeds_types = array_values($feeds_types);
```

Comment:
Will there be other types to support? What about events? Maybe contextual better to call as $feeds_news_types if nothing else.

If other types to come later, will need a loop to dynamically build groupings by type including the labels to not rely on hard-coding strings.

Reply:
Please check the attached screenshot. We are getting same type because all terms belong to same category. Hence we are using the name instead of type.
![image](https://user-images.githubusercontent.com/121072170/222941818-fbb2446e-b546-4088-b963-468d902eac14.png)